src/terminal/cli/select.lua

```diff
@@ -0,0 +1,139 @@
+-- terminal.cli.select
+-- A single-choice interactive menu widget for CLI tools.
+local t = require("terminal")
+local Sequence = require("terminal.sequence")
+local utils = require("terminal.utils")
+
+local Select = utils.class()
+
+-- Key bindings
+local key_names = {
+  ["\27[A"] = "up",    -- Up arrow
+  ["k"]     = "up",    -- Vim-style up
+  ["\27[B"] = "down",  -- Down arrow
+  ["j"]     = "down",  -- Vim-style down
+  ["\r"]    = "enter", -- Carriage return (Enter)
+  ["\n"]    = "enter", -- Newline (Enter)
+  ["\27"]   = "esc",   -- Escape
+}
+
+-- UI symbols
+local diamond = "◇"
+local pipe = "│"
+local circle = "○"
+local dot = "●"
+
+-- Initialize menu
+function Select:init(options)
+  assert(type(options) == "table", "options must be a table")
+  assert(type(options.choices) == "table", "choices must be a table")
+  assert(#options.choices > 0, "choices must not be empty")
+  for _, val in ipairs(options.choices) do
+    assert(type(val) == "string", "each choice must be a string")
+  end
```

Comment:
```suggestion
  assert(type(self.choices) == "table", "choices must be a table")
  assert(#self.choices > 0, "choices must not be empty")
  for _, val in ipairs(self.choices) do
    assert(type(val) == "string", "each choice must be a string")
  end
```

Just replace `options` with `self` should work with the other class model.
Obviously other occurrences must also be adapted.